website/docs/cdktf/examples-and-guides/refactoring.mdx

```diff
@@ -0,0 +1,255 @@
+---
+page_title: Refactoring - CDK for Terraform
+description: >-
+  As your CDKTF application grows in size and complexity, the need to move resources between stacks becomes more common. This guide will walk you through the process of refactoring your stacks.
+---
+
+# Refactoring
+
+As your CDKTF application grows in size and complexity, the need to move resources between stacks becomes more common. This guide will walk you through the process of refactoring your stacks. In general you can change the names of stacks freely without an effect on the synthesized code. If you change the name of a resource this might lead to a re-creation of the resource, especially if you move the resource from one stack to another.
+
```

Comment:
I think that's generally a hard question to answer in a document as short as this. 

That being said, I think we could be a bit more opinionated in our application structure that reduces the need for this kind of refactoring, but that's a conversation for a later time.

Reply:
okay that's fair. Even if it's just a few sentences, I think folks like guidance. But I trust your judgment that it may be too complex for this short space!